### PR TITLE
refactor(backend): unauthenticated user object

### PR DESCRIPTION
**Pull request by Mogge**
Motivation: We have in our resolvers every time we fetch the user from the context a test, that the user exists.

This is because of the type definition of the user in the context, which may be null. To get rid of the if statements, I created an unauthenticated user object with id -1, that is put in the context when the user is not authenticated.
The coverage of branches is now 9% higher, and we do not have that useless code anymore.

Please check for any authentication issues.

### Issues

- None

### Todo

- [X] None

### CODE REVIEW

File: backend/src/context/context.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ const JWKS = createRemoteJWKSet(new URL(CONFIG.JWKS_URI))
 
 export type Context = {
   config: typeof CONFIG
-  user: UserWithProfile | null
+  user: UserWithProfile
   dataSources: { prisma: PrismaClient }
 }
```
Comment (roschaefer) [inline backend/src/context/context.ts:17]: My suggestion is to simply create an:
```ts
export type AuthorizedContext = {
  user: UserWithProfile
// ...
```

And use this in our authorized resolvers:
```ts
  @Authorized()
  @Mutation(() => Int)
  async joinMyTable(@Ctx() context: AuthorizedContext): Promise<number> {
  // ...
```
 That should fix our typescript issues.

**Review by roschaefer — CHANGES_REQUESTED** [review]
sorry

I'm sorry, but this is quite a hack. I like that we want to get rid of the if-clauses though! But I think it's much easier and cleaner, if we simply use a different type for the context argument.

File: backend/src/context/context.ts (hunk 1 above)
Comment (Mogge) [reply backend/src/context/context.ts]: Thanks for the hint. Please have a look again

**Comment by Mogge** [comment]
@roschaefer May you have a look again?

**Review by roschaefer — APPROVED** [review]
typescript

Feel free to incorporate the suggestion or not.

**Pull request merged**